Code/ForceField/UFF/TorsionAngle.cpp

```diff
@@ -192,11 +192,12 @@ double TorsionAngleContrib::getEnergy(double *pos) const {
   double cosNPhi = 0.0;
   switch (d_order) {
     case 2:
-      cosNPhi = cosPhi * cosPhi - sinPhiSq;
+      // cos(2x) = 1 - 2sin^2(x)
+      cosNPhi = 1 - 2 * sinPhiSq;
       break;
     case 3:
-      // cos(3x) = cos^3(x) - 3*cos(x)*sin^2(x)
-      cosNPhi = cosPhi * (cosPhi * cosPhi - 3. * sinPhiSq);
+      // cos(3x) = cos^3(x) - 3*cos(x)*sin^2(x) = 4cos^3(x) -3cos(x)
+      cosNPhi = cosPhi * (4 * cosPhi * cosPhi - 3.);
```

Comment:
I got an x64 (intel) build working on my M1 Mac and can now reproduce the problems observed in the CI builds.
This change is responsible for the failure: the minimization at line 163 of `Code/ForceField/Wrap/testConstraints.py ` fails to converge in the default 200 iterations. If I bump the number of itrations up to 211, then it does pass.
When using the original formulation - `cosNPhi = cosPhi * (cosPhi * cosPhi - 3. * sinPhiSq)`  - the minimization converges in ~150 iterations.

Subtracting a constant from these cosPhi products definitely does change something with the stability of the numerics.


Change is not necessarily bad: it could be that your version is actually more stable and that it's highlighting what was previously a problem that we had not noticed. 

Your comment on the commits that make this change indicate that you were intending to improve the stability of the numerics; do you have some kind off references indicating that the way you restructured these calculations is actually more stable?